### DISCUSSION TITLE
Javascript clashing with quarto-nav.js

### DISCUSSION THREAD

**Discussion by baptiste** [quarto-dev/quarto-cli#3889] [Q&A] [answered]
I'm trying to use the [lightgalleryjs](https://www.lightgalleryjs.com/) library to display collections of photos on my quarto website. Unfortunately, there's a glitch, and after some trial and error I've narrowed it down to a clash with `quarto-nav.js`. If I comment it out manually in the html, the gallery behaves as expected; if I uncomment it, the lightbox view fails to load the second image (spinning wheel). My test webpage is here:

https://baptiste.github.io/quarto-gallery/

with the [manually edited html](https://github.com/baptiste/quarto-gallery/blob/main/docs/index.html#L49)

and here's a screenshot of the bug when quarto-nav is present

<img width="943" alt="Screenshot 2023-01-09 at 10 00 04" src="https://user-images.githubusercontent.com/18970/211218852-22c7b2f4-17d9-4e89-afe4-ab240c34fc7a.png">

I'm not exactly sure of what quarto-nav does, or how/if I can selectively tell quarto not to include it. Alternatively, I'd like to find a way to make the two js scripts coexist peacefully, but I don't know how to isolate the problem (let alone fix it); the web console didn't return an obvious error I could understand.

**Comment by dragonstyle** [comment, marked as answer]
Sorry you've run into this. When I visit the gallery (very nice layout BTW!), I am able to use the gallery feature without any errors or issues. I'm not sure whether you've fixed the issue or something else change or if perhaps the issue is local to your machine (e.g. perhaps a cached version of something)? Anyhow if you are able to still reproduce this I would love to track it down and fix it!

**Reply by baptiste** [reply]
Thanks for checking this and sorry I wasn't very clear: the page I linked to had a manual edit to comment out `quarto.nav.js`, which is how I isolated the problem. The exact same page, with just this one line un-commented, is at https://baptiste.github.io/quarto-gallery/index-bug.html and should show the problem (click on a photo, and the spinning wheel should appear instead of moving to the next photo). The console reveals errors related to `data-src`
, but the field _is_ present in the html so I'm not sure why it disappears when `quarto-nav.js` is included in the page.

<img width="950" alt="Screenshot 2023-01-16 at 15 16 37" src="https://user-images.githubusercontent.com/18970/212584548-fd5be256-53a9-4e3e-bdb1-724c461ccd6d.png">


I quite like this lightgalleryjs (this is straight from one of their examples), as you say it has a nice layout, and lots of features. I'm currently doing [a simpler thing](https://blag.bapt.xyz/posts/gallery/#generating-the-page-with-quarto) with css columns and the built-in lightbox extension on my [website](https://p.bapt.xyz/kaikoura/), but I'd like to switch over to lightgalleryjs eventually.

**Reply by baptiste** [reply]
@dragonstyle should I open an issue?

**Reply by dragonstyle** [reply]
This should be fixed in the next update of Quarto 1.3 (I just started a build with the fix). We were always adding an `href` attribute to document links, which appeared to be confusing lightGallery.

https://github.com/quarto-dev/quarto-cli/commit/1bc73c76bbbe0795bb7938664750114e687b2adf

**Reply by baptiste** [reply]
Nice! Thanks a lot, that did fix it.

**Reply by baptiste** [reply]
The `href=""` seems to be back? I'm only getting back to this photo gallery idea now, and looking at the HTML produced by Quarto (1.4.322) shows this `href=""` being introduced (hard-coded).

**Reply by dragonstyle** [reply]
Can you let me know where you're seeing this? We didn't change the fixed behavior, but it is possible that it leaked in somewhere else...
